tests/models/test_sasrec.py

```diff
@@ -582,3 +593,337 @@ def test_get_dataloader_recommend(
+        cat_features_item_net = CatFeaturesItemNet.from_dataset(ds, n_factors=10, dropout_rate=0.5)
+        assert cat_features_item_net is None
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning")
+class TestItemNetConstructor:
+    def setup_method(self) -> None:
+        self._seed_everything()
+
+    def _seed_everything(self) -> None:
+        torch.use_deterministic_algorithms(True)
+        seed_everything(32, workers=True)
+
+    @pytest.fixture
+    def dataset_item_features(self) -> Dataset:
+        item_features = pd.DataFrame(
+            [
+                [11, "f1", "f1val1"],
+                [11, "f2", "f2val1"],
+                [12, "f1", "f1val1"],
+                [12, "f2", "f2val2"],
+                [13, "f1", "f1val1"],
+                [13, "f2", "f2val3"],
+                [14, "f1", "f1val2"],
+                [14, "f2", "f2val1"],
+                [15, "f1", "f1val2"],
+                [15, "f2", "f2val2"],
+                [17, "f1", "f1val2"],
+                [17, "f2", "f2val3"],
+                [16, "f1", "f1val2"],
+                [16, "f2", "f2val3"],
+                [11, "f3", 0],
+                [12, "f3", 1],
+                [13, "f3", 2],
+                [14, "f3", 3],
+                [15, "f3", 4],
+                [17, "f3", 5],
+                [16, "f3", 6],
+            ],
+            columns=["id", "feature", "value"],
+        )
+        ds = Dataset.construct(
+            INTERACTIONS,
+            item_features_df=item_features,
+            cat_item_features=["f1", "f2"],
+        )
+        return ds
+
+    def test_catalogue(self) -> None:
+        item_net = ItemNetConstructor.from_dataset(
+            DATASET, n_factors=10, dropout_rate=0.5, item_net_block_types=(IdEmbeddingsItemNet,)
+        )
+        expected_feature_catalogue = torch.arange(0, item_net.n_items)
+        assert torch.equal(item_net.catalogue, expected_feature_catalogue)
+
+    @pytest.mark.parametrize(
+        "item_net_block_types,n_factors",
+        (
+            ((IdEmbeddingsItemNet,), 8),
+            ((IdEmbeddingsItemNet, CatFeaturesItemNet), 16),
+        ),
+    )
+    def test_get_all_embeddings(
+        self, dataset_item_features: Dataset, item_net_block_types: tp.Sequence[tp.Type[ItemNetBase]], n_factors: int
+    ) -> None:
+        item_net = ItemNetConstructor.from_dataset(
+            dataset_item_features, n_factors=n_factors, dropout_rate=0.5, item_net_block_types=item_net_block_types
+        )
+        assert item_net.get_all_embeddings().shape == (item_net.n_items, n_factors)
+
+    @pytest.mark.parametrize(
+        "item_net_block_types,sparse_features,expected_n_item_net_blocks",
+        (
+            ((IdEmbeddingsItemNet,), True, 1),
+            ((IdEmbeddingsItemNet, CatFeaturesItemNet), True, 2),
+            ((IdEmbeddingsItemNet,), False, 1),
+            ((IdEmbeddingsItemNet, CatFeaturesItemNet), False, 1),
+        ),
+    )
+    def test_create_from_dataset(
```

Comment:
Please name the test according to what you are testing. Here you test that number of item net block is correct. Let's name it `test_correct_number_of_item_net_blocks`

Reply:
Fixed